<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>main_gui</class>
 <widget class="QMainWindow" name="main_gui">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1031</width>
    <height>597</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>1031</width>
    <height>597</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>网络嗅探器</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="minimumSize">
    <size>
     <width>1031</width>
     <height>552</height>
    </size>
   </property>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>171</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>网络设备列表</string>
    </property>
   </widget>
   <widget class="QPushButton" name="flushDev">
    <property name="geometry">
     <rect>
      <x>180</x>
      <y>0</y>
      <width>171</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>刷新设备列表</string>
    </property>
   </widget>
   <widget class="QPushButton" name="buttStart">
    <property name="geometry">
     <rect>
      <x>360</x>
      <y>0</y>
      <width>81</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>开始</string>
    </property>
   </widget>
   <widget class="QPushButton" name="buttStop">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>480</x>
      <y>0</y>
      <width>81</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>停止</string>
    </property>
    <property name="checkable">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="buttClear">
    <property name="geometry">
     <rect>
      <x>600</x>
      <y>0</y>
      <width>81</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>清空</string>
    </property>
   </widget>
   <widget class="QPushButton" name="buttFilter">
    <property name="geometry">
     <rect>
      <x>730</x>
      <y>0</y>
      <width>81</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>过滤</string>
    </property>
   </widget>
   <widget class="QSplitter" name="splitter">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>30</y>
      <width>1031</width>
      <height>521</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="minimumSize">
     <size>
      <width>1031</width>
      <height>521</height>
     </size>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
    <property name="handleWidth">
     <number>10</number>
    </property>
    <widget class="QListWidget" name="devList">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Fixed" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="minimumSize">
      <size>
       <width>351</width>
       <height>0</height>
      </size>
     </property>
     <property name="font">
      <font>
       <pointsize>12</pointsize>
      </font>
     </property>
    </widget>
    <widget class="QTableWidget" name="packageList">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="minimumSize">
      <size>
       <width>671</width>
       <height>521</height>
      </size>
     </property>
     <property name="sizeAdjustPolicy">
      <enum>QAbstractScrollArea::AdjustToContents</enum>
     </property>
     <property name="editTriggers">
      <set>QAbstractItemView::NoEditTriggers</set>
     </property>
     <property name="dragEnabled">
      <bool>false</bool>
     </property>
     <property name="dragDropOverwriteMode">
      <bool>true</bool>
     </property>
     <property name="alternatingRowColors">
      <bool>true</bool>
     </property>
     <property name="selectionMode">
      <enum>QAbstractItemView::SingleSelection</enum>
     </property>
     <property name="selectionBehavior">
      <enum>QAbstractItemView::SelectRows</enum>
     </property>
     <property name="textElideMode">
      <enum>Qt::ElideLeft</enum>
     </property>
     <property name="verticalScrollMode">
      <enum>QAbstractItemView::ScrollPerPixel</enum>
     </property>
     <property name="horizontalScrollMode">
      <enum>QAbstractItemView::ScrollPerPixel</enum>
     </property>
     <property name="sortingEnabled">
      <bool>true</bool>
     </property>
     <property name="rowCount">
      <number>100</number>
     </property>
     <property name="columnCount">
      <number>8</number>
     </property>
     <attribute name="horizontalHeaderCascadingSectionResizes">
      <bool>true</bool>
     </attribute>
     <attribute name="horizontalHeaderDefaultSectionSize">
      <number>100</number>
     </attribute>
     <attribute name="horizontalHeaderShowSortIndicator" stdset="0">
      <bool>true</bool>
     </attribute>
     <attribute name="horizontalHeaderStretchLastSection">
      <bool>true</bool>
     </attribute>
     <attribute name="verticalHeaderVisible">
      <bool>false</bool>
     </attribute>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <column/>
     <column/>
     <column/>
     <column/>
     <column/>
     <column/>
     <column/>
     <column/>
    </widget>
   </widget>
   <widget class="QPushButton" name="buttScan">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>950</x>
      <y>0</y>
      <width>81</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>端口扫描</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1031</width>
     <height>23</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu123">
    <property name="title">
     <string>菜单</string>
    </property>
   </widget>
   <addaction name="menu123"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="ex4_gui.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>flushDev</sender>
   <signal>clicked(bool)</signal>
   <receiver>main_gui</receiver>
   <slot>flushDev()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>269</x>
     <y>35</y>
    </hint>
    <hint type="destinationlabel">
     <x>438</x>
     <y>80</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttFilter</sender>
   <signal>clicked()</signal>
   <receiver>main_gui</receiver>
   <slot>filter_show()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>761</x>
     <y>32</y>
    </hint>
    <hint type="destinationlabel">
     <x>838</x>
     <y>38</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttStart</sender>
   <signal>clicked()</signal>
   <receiver>main_gui</receiver>
   <slot>start()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>417</x>
     <y>40</y>
    </hint>
    <hint type="destinationlabel">
     <x>451</x>
     <y>38</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttStop</sender>
   <signal>clicked()</signal>
   <receiver>main_gui</receiver>
   <slot>stop()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>534</x>
     <y>35</y>
    </hint>
    <hint type="destinationlabel">
     <x>578</x>
     <y>40</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttClear</sender>
   <signal>clicked()</signal>
   <receiver>main_gui</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>632</x>
     <y>35</y>
    </hint>
    <hint type="destinationlabel">
     <x>696</x>
     <y>37</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>packageList</sender>
   <signal>cellDoubleClicked(int,int)</signal>
   <receiver>main_gui</receiver>
   <slot>data_show(int,int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>587</x>
     <y>186</y>
    </hint>
    <hint type="destinationlabel">
     <x>355</x>
     <y>142</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>devList</sender>
   <signal>currentRowChanged(int)</signal>
   <receiver>main_gui</receiver>
   <slot>start(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>121</x>
     <y>173</y>
    </hint>
    <hint type="destinationlabel">
     <x>174</x>
     <y>37</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttScan</sender>
   <signal>clicked()</signal>
   <receiver>main_gui</receiver>
   <slot>scan_show()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>979</x>
     <y>34</y>
    </hint>
    <hint type="destinationlabel">
     <x>925</x>
     <y>37</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>flushDev()</slot>
  <slot>filter_show()</slot>
  <slot>start(int)</slot>
  <slot>start()</slot>
  <slot>stop()</slot>
  <slot>clear()</slot>
  <slot>data_show(int,int)</slot>
  <slot>scan_show()</slot>
 </slots>
</ui>
